### ISSUE TITLE
PDF export error due to custom HTML tags from plugins

### ISSUE THREAD

**Issue by aicm** [ether/etherpad-lite#2485]
Hello there,

I just realized why HTML to PDF conversion via AbiCommand fails. It's due to custom HTML tags like this:
```HTML
<body>Cool<br><br> Nam fringilla tellus et erat faucibus efficitur. 
Interdum et malesuada fames ac ante ipsum primis in faucibus. 
In mattis tempus aliquet.<br><fs15>ddaa</fs15><br><br></body>
```

The fs15 is not part of the current HTML tags libxml2 understands, and thus throws an error during the conversion request and fails to complete. That tag is generated by the awesome *ep_font_size* plugin by *John M.* I tried to poke around and see if I could swap that to a class="fs15" but unfortunately I don't see anything easy for a non-coder to change. 

Here's the AbiCommand debug log while attempting to convert this HTML file: http://pastie.org/private/mkvwxwqaiy3movoepacugq
```python
DEBUG: !!!!!!!!!!----------------- Created string db8010ac-a312-11e4-9ee0-adcae427221b
DEBUG: !!!!!!!!!!----------------- Orig string db8010ac-a312-11e4-9ee0-adcae427221b
DEBUG: PD_DocumentRDF() this:0x1367880 doc:0x1371580
DEBUG: PD_Document::_importFile... name:file:///tmp/etherpad_export_1459309582.html
DEBUG: PD_DocumentRDF::setupWithPieceTable() m_indexAP:0
DEBUG: PD_Document::importStyles -- file is not plain file
DEBUG: PD_Document::importStyles -- file is not plain file
DEBUG: PD_Document::importStyles -- file is not plain file
DEBUG: PD_Document::importStyles -- file is not plain file
DEBUG: PD_Document::importStyles -- file is not plain file
DEBUG: PD_Document::importStyles -- file is not plain file
DEBUG: pd_Document::setAttrProp: setting dom-dir to ltr
DEBUG: TableHelperStack created document = (nil)
DEBUG: startElement: html, parseState: 0, listType: 0
DEBUG: startElement: head, parseState: 6, listType: 0
DEBUG: startElement: title, parseState: 6, listType: 0
DEBUG: endElement: title, parseState: 14, listType: 0
DEBUG: startElement: meta, parseState: 6, listType: 0
DEBUG: endElement: meta, parseState: 6, listType: 0
DEBUG: startElement: style, parseState: 6, listType: 0
DEBUG: endElement: style, parseState: 6, listType: 0
DEBUG: endElement: head, parseState: 6, listType: 0
DEBUG: startElement: body, parseState: 6, listType: 0
DEBUG: XHTML Import - appendStruxStrux type 0 document 0x1371580
DEBUG: CSS->Props (utf8val): []
DEBUG: XHTML Import - appendStruxStrux type 1 document 0x1371580
DEBUG: CSS->Props (utf8val): []
DEBUG: startElement: br, parseState: 3, listType: 0
DEBUG: endElement: br, parseState: 3, listType: 0
DEBUG: startElement: br, parseState: 3, listType: 0
DEBUG: endElement: br, parseState: 3, listType: 0
DEBUG: startElement: br, parseState: 3, listType: 0
DEBUG: endElement: br, parseState: 3, listType: 0
DEBUG: startElement: br, parseState: 3, listType: 0
DEBUG: endElement: br, parseState: 3, listType: 0
DEBUG: startElement: fs15, parseState: 3, listType: 0
DEBUG: Unknown tag [fs15]
DEBUG: libxml2/html: error: Tag fs15 invalid
endElement: fs15, parseState: 3, listType: 0
DEBUG: Unknown end tag [fs15]
DEBUG: startElement: br, parseState: 3, listType: 0
DEBUG: endElement: br, parseState: 3, listType: 0
DEBUG: startElement: br, parseState: 3, listType: 0
DEBUG: endElement: br, parseState: 3, listType: 0
DEBUG: endElement: body, parseState: 3, listType: 0
DEBUG: XHTML Import - appendStruxStrux type 0 document 0x1371580
DEBUG: CSS->Props (utf8val): []
DEBUG: XHTML Import - appendStruxStrux type 1 document 0x1371580
DEBUG: CSS->Props (utf8val): []
DEBUG: CSS->Props (utf8val): []
DEBUG: XHTML Import - appendStruxStrux type 1 document 0x1371580
DEBUG: CSS->Props (utf8val): []
DEBUG: endElement: html, parseState: 0, listType: 0
DEBUG: Error parsing '' (Line: 39, Column: 8)
DEBUG: Problem reading document
DEBUG: XHTML Import - appendStruxStrux type 0 document 0x1371580
DEBUG: CSS->Props (utf8val): []
DEBUG: XHTML Import - appendStruxStrux type 1 document 0x1371580
DEBUG: CSS->Props (utf8val): []
DEBUG: DOM: suffix: .html
DEBUG: PD_Document::importFile -- could not import file
AbiWord: could not open the file [/tmp/etherpad_export_1459309582.html]
DEBUG: ~PD_DocumentRDF() this:0x1367880
error -1
```
This is a centos 7 x64bit, running the latest git release of etherpad, and abiword 3.0.1 compiled from source. Latest svn abiword, and 2.9.4 fail the same way.

**Comment by JohnMcLear** [comment]
Ugh yeah we need that block so it maintains formatting when editing.  The fix will be to edit the html on export
----- Reply message -----

**Comment by JohnMcLear** [comment]
This actually needs posting on the ep_page_view repo not here, just FYI

**Comment by JohnMcLear** [comment]
Just FWIW the PR is in, waiting on merge and release.  I can publish fix to the apps but you will need to run Etherpad develop to test them now.

**Comment by aicm** [comment]
My bad about the cross post. I wasn't aware of the github for the plugin. Thank you!
